examples/main.tf

```diff
@@ -60,28 +60,25 @@ resource "materialize_secret" "example_secret" {
 
 
 # Create SSH Connection
-resource "materialize_connection" "example_ssh_connection" {
+resource "materialize_connection_ssh_tunnel" "example_ssh_connection" {
   name            = "ssh_example_connection"
   schema_name     = "public"
-  connection_type = "SSH TUNNEL"
-  ssh_host        = "example.com"
-  ssh_port        = 22
-  ssh_user        = "example"
+  host            = "example.com"
+  port            = 22
+  user            = "example"
 }
 
 # # Create a AWS Private Connection
 # Note: you need the max_aws_privatelink_connections increased for this to work:
 # show max_aws_privatelink_connections;
-resource "materialize_connection" "privatelink_conn" {
-  name                               = "privatelink_conn"
-  schema_name                        = "public"
-  connection_type                    = "AWS PRIVATELINK"
-  aws_privatelink_service_name       = "com.amazonaws.us-east-1.materialize.example"
-  aws_privatelink_availability_zones = ["use1-az2", "use1-az1"]
+resource "materialize_connection_aws_privatelink" "privatelink_conn" {
```

Comment:
For consistency maybe name it `"materialize_connection_aws_privatelink" "example_aws_privatelink_conn"`